### PR TITLE
[CIT-338] Split certificate module to multiple files

### PR DESCRIPTION
**Pull request by makr11st**
## Proposed changes

This code splits certificate module into its own module directory with proposed structure.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: mapper/thin_edge_json/Cargo.toml
Comment (makr11st) [inline mapper/thin_edge_json/Cargo.toml:16]: cargo fmt

File: tedge/src/services/mod.rs
Comment (makr11st) [inline tedge/src/services/mod.rs:200]: clippy

File: tedge/src/services/mod.rs
Comment (makr11st) [inline tedge/src/services/mod.rs:219]: clippy

File: tedge/src/cli/mod.rs
Comment (PradeepKiruvale) [inline tedge/src/cli/mod.rs]: remove this commented code

**Review by PradeepKiruvale — COMMENTED** [review]
LGTM,  Is not it better to move the tests also into another separate file. Like tests.rs or something like that?

File: tedge/src/cli/certificate/cli.rs
Comment (mneumann) [reply tedge/src/cli/certificate/cli.rs:38]: You might not need `config_repository` in all commands, as now Didier derives the `device.id` within the `TEdgeConfig`, so no need to serialize `device.id`.

File: tedge/src/cli/certificate/cli.rs
Comment (mneumann) [inline tedge/src/cli/certificate/cli.rs:79]: I'd rather move the `build_command` implementation into the `BuildCommand` for `TEdgeCertCli`. This avoids loading the config twice.

File: tedge/src/cli/certificate/remove.rs
Comment (mneumann) [inline tedge/src/cli/certificate/remove.rs:42]: this will go away after #190

**Review by mneumann — COMMENTED** [review]
Please wait until PR #190 is merged. And rebase. Hope you'll not have too much fallout.

**Review by didier-wenzek — APPROVED** [review]
Nice to see the final result of this command refactoring.

**Pull request merged**
